pkg/standalone-utils/contracts/relayer/EulerWrapping.sol

```diff
@@ -0,0 +1,126 @@
+    using SafeERC20 for IERC20;
+    using FixedPoint for uint256;
+
+    address private immutable _eulerProtocol;
+
+    /**
+     * @dev Euler protocol needs to be approved to transer mainToken
+     * @param eulerProtocol - the address of the euler protocol
+     */
+    constructor(address eulerProtocol) {
+        _eulerProtocol = eulerProtocol;
+    }
+
+    function wrapEuler(
+        IEulerToken wrappedToken,
+        address sender,
+        address recipient,
+        uint256 amount,
+        uint256 outputReference
+    ) external payable {
+        if (_isChainedReference(amount)) {
+            amount = _getChainedReferenceValue(amount);
+        }
+
+        IERC20 underlying = IERC20(wrappedToken.underlyingAsset());
+
+        // The wrap caller is the implicit sender of tokens, so if the goal is for the tokens
+        // to be sourced from outside the relayer, we must first pull them here.
+        if (sender != address(this)) {
+            require(sender == msg.sender, "Incorrect sender");
+            _pullToken(sender, underlying, amount);
+        }
+
+        underlying.safeApprove(_eulerProtocol, amount);
+
+        // Deposit MainToken into EulerToken
+        // 0 for the Euler primary account
+        wrappedToken.deposit(0, amount);
+
+        uint256 withdrawnWrappedAmount = wrappedToken.balanceOf(address(this));
+
+        if (recipient != address(this)) {
+            IERC20(wrappedToken).safeApprove(address(this), withdrawnWrappedAmount);
+            IERC20(wrappedToken).safeTransferFrom(address(this), recipient, withdrawnWrappedAmount);
+        }
+
+        if (_isChainedReference(outputReference)) {
+            _setChainedReferenceValue(outputReference, withdrawnWrappedAmount);
+        }
+    }
+
+    function unwrapEuler(
+        IEulerToken wrappedToken,
+        address sender,
+        address recipient,
+        uint256 amount,
+        uint256 outputReference
+    ) external payable {
+        if (_isChainedReference(amount)) {
+            amount = _getChainedReferenceValue(amount);
+        }
+
+        // The unwrap caller is the implicit sender of tokens, so if the goal is for the tokens
+        // to be sourced from outside the relayer, we must first pull them here.
+        if (sender != address(this)) {
+            require(sender == msg.sender, "Incorrect sender");
+            _pullToken(sender, IERC20(address(wrappedToken)), amount);
+        }
+
+        IERC20 mainToken = IERC20(wrappedToken.underlyingAsset());
+
+        // Euler offers two ways to withdraw. Either calculate the MainTokenOut via
+        // 1. MainTokenOut = wrappedToken.convertBalanceToUnderlying(WrappedTokenAmount); or
+        // 2. withdraw as Much MainToken as one gets when depositing WrappedTokenAmount back
+        // When using option 1, the possibility exists to have dust of WrappedToken in the relayer
+        // Therefor option 2 is chosen
+        // 0 for the Euler primary account
```

Comment:
This wording is a bit unclear to me. I would change it like this. Also minor grammatical fixes.
```suggestion
        // Euler offers two ways to withdraw:
        //     1. Calculate mainTokenOut via wrappedToken.convertBalanceToUnderlying(wrappedTokenAmount)
        //     2. Redeem the account's full balance of wrappedToken for mainToken
        // Option 1 may leave wrappedToken dust in the relayer, so we choose option 2
        // The 0 argument is for the Euler primary account
```